src/main/kotlin/org/arend/intention/SwitchParamImplicitnessIntention.kt

```diff
@@ -0,0 +1,25 @@
+package org.arend.intention
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import org.arend.psi.*
+import org.arend.psi.ext.ArendCompositeElement
+import org.arend.util.ArendBundle
+
+class SwitchParamImplicitnessIntention : SelfTargetingIntention<ArendCompositeElement> (ArendCompositeElement::class.java, ArendBundle.message("arend.coClause.switchParamImplicitness")) {
+    override fun isApplicableTo(element: ArendCompositeElement, caretOffset: Int, editor: Editor): Boolean {
+        return when (element) {
+            is ArendNameTele, is ArendLamTele, is ArendFieldTele, is ArendTypeTele -> true
+            else -> false
+        }
+    }
+
+    override fun applyTo(element: ArendCompositeElement, project: Project, editor: Editor) {
+        val range = element.textRange
+        val openBracket = if (editor.document.text[range.startOffset] == '(') "{" else "("
+        val closeBracket = if (editor.document.text[range.endOffset - 1] == ')') "}" else ")"
```

Comment:
I don't think it's a good idea to use the document API here. I suggest checking the `PsiElement`s, and use `PsiElement::replace` instead